src/windows/models/properties_model.py

```diff
@@ -305,109 +305,98 @@ def color_update(self, item, new_color, interpolation=-1, interpolation_details=
+        for color, new_value in [
+                ("red", new_color.red()),
+                ("blue", new_color.blue()),
+                ("green", new_color.green()),
+                ]:
+
+            # Keyframe
+            # Loop through points, find a matching points on this frame
+            found_point = False
+            for point in d[property_key].get(color, {}).get("Points", []):
+                log.debug("looping points: co.X = %s" % point["co"]["X"])
+                if interpolation == -1 and point["co"]["X"] == self.frame_number:
+                    # Found point, Update value
+                    found_point = True
+                    clip_updated = True
+                    # Update point
+                    point["co"]["Y"] = new_value
+                    log.debug(
+                        "updating point: co.X = %d to value: %.3f",
+                        point["co"]["X"], float(new_value))
+                    break
+
+                elif interpolation > -1 and point["co"]["X"] == previous_point_x:
+                    # Only update interpolation type (and the LEFT side of the curve)
+                    found_point = True
+                    clip_updated = True
+                    point["interpolation"] = interpolation
+                    if interpolation == 0:
+                        point["handle_right"] = point.get("handle_right") or {"Y": 0.0, "X": 0.0}
+                        point["handle_right"]["X"] = interpolation_details[0]
+                        point["handle_right"]["Y"] = interpolation_details[1]
+
+                    log.debug(
+                        "updating interpolation mode point: co.X = %d to %d",
+                        point["co"]["X"], interpolation)
+                    log.debug("use interpolation preset: %s", str(interpolation_details))
+
+                elif interpolation > -1 and point["co"]["X"] == closest_point_x:
+                    # Only update interpolation type (and the RIGHT side of the curve)
+                    found_point = True
+                    clip_updated = True
+                    point["interpolation"] = interpolation
+                    if interpolation == 0:
+                        point["handle_left"] = point.get("handle_left") or {"Y": 0.0, "X": 0.0}
+                        point["handle_left"]["X"] = interpolation_details[2]
+                        point["handle_left"]["Y"] = interpolation_details[3]
+
+                    log.debug(
+                        "updating interpolation mode point: co.X = %d to %d",
+                        point["co"]["X"], interpolation)
+                    log.debug("use interpolation preset: %s", str(interpolation_details))
+
+            # Create new point (if needed)
+            if not found_point:
+                clip_updated = True
+                log.debug("Created new point at X=%d", self.frame_number)
+                d[property_key].setdefault(color, {}).setdefault("Points", []).append({
```

Comment:
Looks like as well as cleaning up, you addressed the safety issue my PR was aimed at :+1:

Reply:
Yeah, along the way I kind of did incorporate all of the excellent fixes you'd already made. I was already in there, and they were fresh in my mind! :laughing: